### ISSUE TITLE
error 429 on connection fo all users

### ISSUE THREAD

**Issue by leccyril** [matrix-org/synapse#15477]
### Description

could not connect anymore from hydrogen sdk or element on matrix server

we have matrix server running from 10/20h then no specific error, and no possibility to connect anymore with any account

We have one script each day will recreate all rooms (if new comer joind society), i suspect it to make it crash, but before it was working (we just updated matrix version)

![image](https://user-images.githubusercontent.com/3234598/233918202-0693d322-ca8b-448d-8bd5-72bc81c7ba2b.png)

![image](https://user-images.githubusercontent.com/3234598/233918295-2adc14a5-5101-4e1d-b6ed-85fdb41d4400.png)

### Steps to reproduce

-launch matrix server and wait

### Homeserver

homeserver

### Synapse Version

{"server":{"name":"Synapse","version":"1.81.0"}}

### Installation Method

Docker (matrixdotorg/synapse)

### Database

postgres

### Workers

Single process

### Platform

docker on google cloud

### Relevant log output

![image](https://user-images.githubusercontent.com/3234598/233918898-7a750792-1cd2-4792-b521-f25ea0c8fec7.png)

```shell
2023-04-24 06:28:02,496 - synapse.http.server - 751 - WARNING - GET-403922 - Not sending response to request <XForwardedForRequest at 0x7f6b19dcea10 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745162_0_27457_469695_9_13513_495267_0_1&timeout=30000&filter ││ 2023-04-24 06:28:02,583 - synapse.http.server - 751 - WARNING - GET-403673 - Not sending response to request <XForwardedForRequest at 0x7f6b103f6510 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745162_0_27457_469695_9_13513_495260_0_1&timeout=30000&filter ││ 2023-04-24 06:28:53,724 - synapse.handlers.auth - 1068 - WARNING - POST-404682 - Attempted to login as @valérie.clément.at.properti.es:matrix.properti.es but they do not exist                                                                                             ││ 2023-04-24 06:28:53,724 - synapse.module_api - 709 - WARNING - POST-404682 - Using deprecated ModuleApi.register which creates a dummy user device.                                                                                                                         ││ 2023-04-24 06:28:53,727 - synapse.handlers.auth - 2164 - WARNING - POST-404682 - Failed to run module API callback <function load_single_legacy_password_auth_provider.<locals>.async_wrapper.<locals>.wrapped_check_password at 0x7f6b1cf70e00>: 400: User ID can only con ││ 2023-04-24 06:28:53,740 - synapse.handlers.auth - 1068 - WARNING - POST-404682 - Attempted to login as @valérie.clément.at.properti.es:matrix.properti.es but they do not exist                                                                                             ││ 2023-04-24 06:28:53,760 - synapse.api.auth - 448 - WARNING - GET-404697 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:24,560 - synapse.api.auth - 448 - WARNING - GET-404909 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:27,649 - synapse.api.auth - 448 - WARNING - GET-404919 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:30,735 - synapse.api.auth - 448 - WARNING - GET-404925 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:33,821 - synapse.api.auth - 448 - WARNING - GET-404950 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:36,893 - synapse.api.auth - 448 - WARNING - GET-404970 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:41,907 - synapse.api.auth - 448 - WARNING - GET-405001 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:45,026 - synapse.api.auth - 448 - WARNING - GET-405036 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:48,128 - synapse.api.auth - 448 - WARNING - GET-405068 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:29:51,191 - synapse.api.auth - 448 - WARNING - GET-405076 - Invalid access token in auth: <class 'pymacaroons.exceptions.MacaroonDeserializationException'> cannot determine data format of binary-encoded macaroon.                                          ││ 2023-04-24 06:30:51,752 - synapse.http.server - 751 - WARNING - GET-405332 - Not sending response to request <XForwardedForRequest at 0x7f6b0a4c2850 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745226_0_27457_469698_9_13513_495272_0_1&timeout=30000&filter ││ 2023-04-24 06:31:04,349 - synapse.http.server - 751 - WARNING - GET-405377 - Not sending response to request <XForwardedForRequest at 0x7f6b08f8fed0 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745227_0_27457_469698_9_13513_495274_0_1&timeout=30000&filter ││ 2023-04-24 06:31:26,045 - synapse.http.server - 751 - WARNING - GET-405487 - Not sending response to request <XForwardedForRequest at 0x7f6b1731a650 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745225_0_27457_469698_9_13513_495272_0_1&timeout=30000&filter ││ 2023-04-24 06:31:26,078 - synapse.http.server - 751 - WARNING - GET-405712 - Not sending response to request <XForwardedForRequest at 0x7f6b11fdac50 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745250_0_27457_469698_9_13513_495277_0_1&timeout=30000&filter ││ 2023-04-24 06:31:40,303 - synapse.http.server - 751 - WARNING - GET-405682 - Not sending response to request <XForwardedForRequest at 0x7f6b1ab3f790 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745249_0_27457_469698_9_13513_495276_0_1&timeout=30000&filter ││ 2023-04-24 06:32:15,170 - synapse.http.server - 751 - WARNING - GET-406081 - Not sending response to request <XForwardedForRequest at 0x7f6b0a6e13d0 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745249_0_27457_469698_9_13513_495277_0_1&timeout=30000&filter ││ 2023-04-24 06:32:18,764 - synapse.http.server - 751 - WARNING - GET-406280 - Not sending response to request <XForwardedForRequest at 0x7f6b0a6c4650 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745250_0_27457_469699_9_13513_495279_0_1&timeout=30000&filter ││ 2023-04-24 06:32:18,768 - synapse.http.server - 751 - WARNING - POST-406463 - Not sending response to request <XForwardedForRequest at 0x7f6b095aee90 method='POST' uri='/_matrix/client/r0/login' clientproto='HTTP/1.1' site='8008'>, already disconnected.               ││ 2023-04-24 06:32:44,808 - synapse.media._base - 265 - WARNING - GET-406979 - Not sending response to request <XForwardedForRequest at 0x7f6b09145f50 method='GET' uri='/_matrix/media/r0/thumbnail/matrix.properti.es/itDeIOMXfGNsssiOYAsbuJSg?width=29&height=29&method=cr ││ 2023-04-24 06:32:44,827 - synapse.media._base - 265 - WARNING - GET-406981 - Not sending response to request <XForwardedForRequest at 0x7f6b04ff5a90 method='GET' uri='/_matrix/media/r0/thumbnail/matrix.properti.es/vQZfPxvAZUhtzdeNiwOvKNwP?width=29&height=29&method=cr ││ 2023-04-24 06:33:15,010 - synapse.http.server - 751 - WARNING - GET-406984 - Not sending response to request <XForwardedForRequest at 0x7f6b095faf90 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745275_0_27457_469701_9_13513_495288_0_1&timeout=30000&filter ││ 2023-04-24 06:33:39,061 - synapse.http.server - 751 - WARNING - GET-407305 - Not sending response to request <XForwardedForRequest at 0x7f6b19d19110 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745298_0_27457_469702_9_13513_495288_0_1&timeout=30000&filter ││ 2023-04-24 06:33:58,567 - synapse.http.server - 751 - WARNING - GET-407470 - Not sending response to request <XForwardedForRequest at 0x7f6b08ef0b90 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745298_0_27457_469702_9_13513_495290_0_1&timeout=30000&filter ││ 2023-04-24 06:34:09,670 - synapse.http.server - 751 - WARNING - GET-407603 - Not sending response to request <XForwardedForRequest at 0x7f6b0f5e9410 method='GET' uri='/_matrix/client/r0/sync?filter=0&timeout=30000&since=s172762_6745298_0_27457_469702_9_13513_495292_0 ││ 2023-04-24 06:34:26,541 - synapse.http.server - 751 - WARNING - GET-407860 - Not sending response to request <XForwardedForRequest at 0x7f6b0926f450 method='GET' uri='/_matrix/client/r0/sync?filter=0&timeout=30000&since=s172762_6745299_0_27457_469703_9_13513_495292_0 ││ 2023-04-24 06:34:57,157 - synapse.http.server - 751 - WARNING - GET-408159 - Not sending response to request <XForwardedForRequest at 0x7f6b14916dd0 method='GET' uri='/_matrix/client/r0/sync?filter=0&timeout=30000&since=s172762_6745324_0_27457_469704_9_13513_495292_0 ││ 2023-04-24 06:34:57,158 - synapse.http.server - 751 - WARNING - GET-408300 - Not sending response to request <XForwardedForRequest at 0x7f6b0bd2c910 method='GET' uri='/_matrix/client/r0/sync?filter=0&timeout=30000&since=s172762_6745324_0_27457_469704_9_13513_495292_0 ││ 2023-04-24 06:36:14,176 - synapse.http.server - 751 - WARNING - GET-408944 - Not sending response to request <XForwardedForRequest at 0x7f6b18ff7050 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745348_0_27457_469705_9_13513_495294_0_1&timeout=30000&filter ││ 2023-04-24 06:36:37,248 - synapse.http.server - 751 - WARNING - GET-409243 - Not sending response to request <XForwardedForRequest at 0x7f6b1650bd90 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745348_0_27457_469706_9_13513_495296_0_1&timeout=30000&filter ││ 2023-04-24 06:37:58,142 - synapse.http.server - 751 - WARNING - GET-410449 - Not sending response to request <XForwardedForRequest at 0x7f6b0d9d03d0 method='GET' uri='/_matrix/client/r0/sync?timeout=0&filter=0&_cacheBuster=5840789774715235' clientproto='HTTP/1.1' sit ││ 2023-04-24 06:38:23,216 - synapse.http.server - 751 - WARNING - GET-410372 - Not sending response to request <XForwardedForRequest at 0x7f6b09053510 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745394_0_27457_469708_9_13513_495296_0_1&timeout=30000&filter ││ 2023-04-24 06:39:21,075 - synapse.http.server - 751 - WARNING - GET-411200 - Not sending response to request <XForwardedForRequest at 0x7f6b05cdfb50 method='GET' uri='/_matrix/client/r0/sync?since=s172762_6745420_0_27457_469709_9_13513_495302_0_1&timeout=30000&filter ││ 2023-04-24 06:41:38,530 - synapse.http.server - 751 - WARNING - GET-412853 - Not sending response to request <XForwardedForRequest at 0x7f6b02d4cc90 method='GET' uri='/_matrix/client/r0/sync?since=s172768_6745467_24_27457_469723_9_13513_495306_0_1&timeout=30000&filte ││ 2023-04-24 06:42:09,351 - synapse.http.server - 751 - WARNING - GET-413206 - Not sending response to request <XForwardedForRequest at 0x7f6b05503650 method='GET' uri='/_matrix/client/r0/sync?since=s172768_6745490_24_27457_469724_9_13513_495307_0_1&timeout=30000&filte │
```

### Anything else that would be useful to know?
Is there a way to delete a user (completely and not just deactivate ?) and ? or change login name to delete accent ?)

**Comment by leccyril** [comment]
![image](https://user-images.githubusercontent.com/3234598/233935847-474cec50-9c3f-466b-b548-b724d838f56f.png)

**Comment by leccyril** [comment]
And now after few minutes after restarting it come back again, so we can't do anything now

![image](https://user-images.githubusercontent.com/3234598/233939937-b0fa2c83-5dca-443c-806c-896dc3fdf123.png)

**Comment by DMRobertson** [comment]
We're using Github's issue tracker for bug reports and feature requests. For this kind of support request, please ask in the [Synapse Admins](https://matrix.to/#/#synapse:matrix.org) room. If your homeserver is inoperable at the moment, you can sign up for an account on the Matrix.org homeserver in order to reach the support room.

Having said that: 

> We have one script each day will recreate all rooms (if new comer joind society), i suspect it to make it crash

A script that rapidly makes lots of Matrix API calls sounds like the kind of thing that may trigger rate limiting.

Aside: It sounds odd that you need to recreate rooms whenever new users appear; instead I would expect your scripts to invite that user to existing rooms, or possibly [get Synapse to do that for you](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#auto_join_rooms).

> , but before it was working (we just updated matrix version)

You could try downgrading to the previous version to see if that fixes the problem. If it does, please let us know as it would suggest a regression in Synapse. However, I would not expect that to be the case, because

> no possibility to connect anymore with **any** account

(emphasis mine.) Synapse's rate limiting is targetted to specific users, IP addresses or rooms; AFAIK there is no global rate limit that would affect all users' logins.

> docker on google cloud

Is this just the Synapse container published by the matrix.org team, or is this a third-party container? Is it running with other services e.g. via docker-compose?

I would check the cloud provider's logs to see if they have any error messages or rate limiting warnings that have been triggered.

From your logs:

> Invalid access token in auth

This doesn't sound good. Is this error new after the upgrade?

> 
![image](https://user-images.githubusercontent.com/8614563/233974881-5c334eaf-f120-4e3f-8147-127b91f0ae68.png)

It looks like something doesn't like the é in your username here. Again, is this error new after the upgrade?

(Please provide logs as text, so that they can be searched and more easily copy-pasted.)

> 
![image](https://user-images.githubusercontent.com/8614563/233975182-a9adc446-3528-4400-bb04-3e8f5791e846.png)

These logs lines demonstrate that you are using _some_ kind of reverse proxy (whose logs you should investigate for rate limiting/429 messages). Additionally, the phrase `Not sending response to request` is only omitted when the other end of the connection has disconnected, i.e. when there is nothing for Synapse to return a response to. Again, reverse proxy logs are the place to look for clues.

Good luck.

**Comment by leccyril** [comment]
Thank you for answer, 

first we made a fix to not allow connection with accent (it is automatic connection) (we will check) 

second yes we will check proxy logs maybe some are catched overthere

For the room because tell me if i am wrong but for each user we have to create room between one user and one other user in the company ?

We get one user and all colleagues and recreate rooms for each user between him and colleagues, indeed like a classic chat when ou click on user on the left side they are already invited and accepted, just to talk (nothing to don on any side), and we prefer to make it on application launch. There is no other room only between each user. so potentially xn rooms.

For the 429 we have it on all accounts but through one application , you think the IP took is the coomon application ?

we have one website, and inside is embedded the chat where each user connect.

it is not the good way to do it ?

**Comment by leccyril** [comment]
We just tried the version 7.8 and version 8.1 of synapse, before the rate limite ask to retry in 3s , now the retry wait 5mn ! i think there is an issue with the retry limit time changed a lot !

**Comment by DMRobertson** [comment]
> it means a user could not see all users saved in matrix.

>  indeed like a classic chat when ou click on user on the left side they are already invited and accepted, just to talk (nothing to don on any side), and we prefer to make it on application launch. 

FWIW Matrix offers a [user directory search endpoint](https://spec.matrix.org/v1.6/client-server-api/#user-directory) which should provide this, provided you [configure `search_all_users: true` in Synapse's config](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#user_directory). This won't automatically _create_ rooms though: clients would still have to create them and invite users.

> For the 429 we have it on all accounts but through one application , you think the IP took is the coomon application ?, so is matrix or synapse could use the header Xforwardheader ?

Sorry, I'm not sure what you're asking here. Can you rephrase?

**Comment by leccyril** [comment]
> > it means a user could not see all users saved in matrix.
> 
> > indeed like a classic chat when ou click on user on the left side they are already invited and accepted, just to talk (nothing to don on any side), and we prefer to make it on application launch.
> 
> FWIW Matrix offers a [user directory search endpoint](https://spec.matrix.org/v1.6/client-server-api/#user-directory) which should provide this, provided you [configure `search_all_users: true` in Synapse's config](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#user_directory). This won't automatically _create_ rooms though: clients would still have to create them and invite users.
> 

OK i will try

> > For the 429 we have it on all accounts but through one application , you think the IP took is the coomon application ?, so is matrix or synapse could use the header Xforwardheader ?
> 
> Sorry, I'm not sure what you're asking here. Can you rephrase?

we use hydrogen SDK on our webapp, i think it is for this reason if one user have error, all users hav same error because ip of webapp is used instead of client IP

maybe we will try this : 

https://github.com/matrix-org/synapse/blob/develop/docs/reverse_proxy.md

We just tried the version 1.7.8 and version 1.8.1 of synapse, before the rate limite ask to retry in 3s , now the retry wait 5mn ! i think there is an issue with the retry limit time changed a lot ! i wrote on synapse room but there are busy with other stuff !

**Comment by DMRobertson** [comment]
> We just tried the version 1.7.8 and version 1.8.1 of synapse, before the rate limite ask to retry in 3s , now the retry wait 5mn ! i think there is an issue with the retry limit time changed a lot ! i wrote on synapse room but there are busy with other stuff !

Those are very old versions with known security flaws. Do you mean 1.78 and 1.81? I don't think there were any rate limiting changes between those releases.

You might also be interested in Synapse's [rate limiting options](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#ratelimiting)—though as mentioned above, I don't think there's evidence in the logs you've provided that Synapse is generating 429 responses.

**Comment by leccyril** [comment]
> > We just tried the version 1.7.8 and version 1.8.1 of synapse, before the rate limite ask to retry in 3s , now the retry wait 5mn ! i think there is an issue with the retry limit time changed a lot ! i wrote on synapse room but there are busy with other stuff !
> 
> Those are very old versions with known security flaws. Do you mean 1.78 and 1.81? I don't think there were any rate limiting changes between those releases.
> 

yes i mean 1.78 and 1.81

> You might also be interested in Synapse's [rate limiting options](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#ratelimiting)—though as mentioned above, I don't think there's evidence in the logs you've provided that Synapse is generating 429 responses.

we change thoses configuration and it was working now, and new version 1.82 come back to 5 s retry waiting time for connection thank you

my last question is we are using kubernetes and traeffic for loadbalancing we don't use any specific reverse proxy, can we specify the X-forwarded-for etc for take the original IP ?

**Comment by leccyril** [comment]
i will close it because now my questions are for other topics

**Comment by leccyril** [comment]
just to confirm that with onfo level log we noticed that ip took was not the client IP but server ip, but we use xforwardedfor in matrix (setting to true) so i wonder was it is not taken
